database/rrdset.c

```diff
@@ -769,6 +769,21 @@ RRDSET *rrdset_create_custom(
 
     rrdsetcalc_link_matching(st);
     rrdcalctemplate_link_matching(st);
+#ifdef ENABLE_DBENGINE
+    BUFFER *object = buffer_create(512);
+    buffer_sprintf(object, "%s/%s", host->machine_guid, st->id);
```

Comment:
This seems a bit heavy - is it not cheaper to just `strlen(st->id)`, add the fixed length of the guid to that and `mallocz` the memory here?

Reply:
Yes it is cheaper, not by much though. In the common case `buffer_create()` will just be a `malloc()` and you won't have to explicitly call `strlen()`. It's probably not a huge difference.

In both cases the `sprintf()` will trigger `strlen()` calls anyways. I'm okay either way, although I prefer `char *` instead of `BUFFER *` optically.